src/query/functions/src/scalars/timestamp/src/interval.rs

```diff
@@ -296,4 +297,72 @@ fn register_number_to_interval(registry: &mut FunctionRegistry) {
             output.push(res);
         }),
     );
+
+    registry.register_passthrough_nullable_1_arg::<IntervalType, Int64Type, _, _>(
+        "to_year",
+        |_, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_1_arg::<IntervalType, Int64Type>(|val, output, _| {
+            output.push(val.months() as i64 / 12);
+        }),
+    );
+    registry.register_passthrough_nullable_1_arg::<IntervalType, Int64Type, _, _>(
+        "to_month",
+        |_, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_1_arg::<IntervalType, Int64Type>(|val, output, _| {
+            output.push(val.months() as i64);
+        }),
+    );
+    // Directly return interval days. Extract need named to_day_of_month
+    registry.register_passthrough_nullable_1_arg::<IntervalType, Int64Type, _, _>(
+        "to_day_of_month",
+        |_, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_1_arg::<IntervalType, Int64Type>(|val, output, _| {
+            output.push(val.days() as i64);
```

Comment:
it's not day of month.

Do you mean `to_day` ?

Reply:
Like the comment : in https://github.com/datafuselabs/databend/blob/ebca5b164b3aaf29fc19d633cf58a574f6aaacac/src/query/sql/src/planner/semantic/type_check.rs#L2945 IntervalKind::Day will convert to call function that named `to_day_of_month`. So this is ann easy way to impl extract day from intervaltype.